File: algorithms/ar-isrt/java/InsertionSort.java
class InsertionSort {
    public static void printArray(int arr[]) {
        int n = arr.length;
        for (int i = 0; i < n; ++i)
            System.out.print(arr[i] + " ");
    }

    public static void main(String args[]) {
        int arr[] = {18, 11, 15, 5, 7};

        InsertionSort ob = new InsertionSort();
        ob.sort(arr);

        printArray(arr);
    }

    public void sort(int arr[]) {
        int n = arr.length;
        int j;
        for (int i = 1; i < n; i++) {
            int key = arr[i];

            for (j = i - 1; (j >= 0 && key < arr[j]); j--)
                arr[j + 1] = arr[j];

            arr[j + 1] = key;
        }

    }
}

File: algorithms/ar-isrt/c/insertion_sort.c
# include <stdio.h>

// Insertion sort function
void insertionSort(int arr[], int len) {
	int i, key;
	for (i=1 ; i<len ; i++) {
		int j = i-1;
		key = arr[i];
		while (j>=0 && arr[j]>key) {
			arr[j+1] = arr[j];
			j-=1;
		}
		arr[j+1] = key;
	}
}

// Function to print an array
void printArray(int arr[], int len) {
	int i;
	for (i=0 ; i<len ; i++)
		printf("%d ", arr[i]);\
	printf("\n");
}

// Function to test insetion sort
int main() {
	int arr[] = {22, 11, 8, 5, 30};
	int len = sizeof(arr)/sizeof(arr[0]);
	insertionSort(arr, len);
	printArray(arr, len);
	return 0;
}